qcoro/qcorotask.h

```diff
@@ -730,6 +730,36 @@ inline T waitFor(QCoro::Task<T> &&task) {
     return detail::waitFor<T>(task);
 }
 
+namespace detail {
+
+template <typename T>
+concept TaskConvertible = requires(T v, TaskPromiseBase t)
+{
+    { t.await_transform(v) };
+};
+
+template<typename T>
+struct awaitable_return_type {
+  using type = decltype(std::declval<T>().await_resume());
+};
+
+template<QCoro::detail::has_operator_coawait T>
+struct awaitable_return_type<T> {
+    using type = typename awaitable_return_type<decltype(std::declval<T>().operator co_await())>::type;
+};
+
+template <typename Awaitable>
+requires TaskConvertible<Awaitable>
+using AwaitableReturnType = typename detail::awaitable_return_type<decltype(std::declval<TaskPromiseBase>().await_transform(Awaitable()))>::type;
```

Comment:
Please use `awaitable_return_type_t` as name of the typedef instead of `AwaitableReturnType` to be consistent with the standard library.